sympy/tensor/tensor_methods.py

```diff
@@ -0,0 +1,477 @@
+# -*- coding: utf-8 -*-
+
+from itertools import permutations
+from sympy.tensor.arraypy import Arraypy, Tensor, copy
+from random import randint
+from sympy.functions.combinatorial.factorials import factorial
+from sympy import expand
+
+
+def symmetric(in_arr):
+    """
+    Creates the symmetric form of input tensor.
+    Input: Arraypy or Tensor with equal axes (array shapes).
+    Output: symmetric array. Output type - Arraypy or Tensor, depends of input
+
+    Examples
+    ========
+
+    >>> from sympy.tensor.arraypy import list2arraypy
+    >>> from sympy.tensor.tensor_methods import symmetric
+    >>> a = list2arraypy(list(range(9)), (3,3))
+    >>> b = symmetric(a)
+    >>> print (b)
+    0  2.00000000000000  4.00000000000000
+    2.00000000000000  4.00000000000000  6.00000000000000
+    4.00000000000000  6.00000000000000  8.00000000000000
+    """
+    if not isinstance(in_arr, Arraypy):
+        raise TypeError('Input must be Arraypy or Tensor type')
+
+    flag = 0
+    for j in range(0, in_arr.rank):
+        if (in_arr.shape[0] != in_arr.shape[j]):
+            raise ValueError('Different size of arrays axes')
+
+    # forming list of tuples for Arraypy constructor of type a = Arraypy( [(a,
+    # b), (c, d), ... , (y, z)] )
+    arg = [(in_arr.start_index[i], in_arr.end_index[i])
+           for i in range(in_arr.rank)]
+
+    # if in_arr Tensor, then res_arr will be Tensor, else it will be Arraypy
+    if isinstance(in_arr, Tensor):
+        res_arr = Tensor(Arraypy(arg), in_arr.ind_char)
+    else:
+        res_arr = Arraypy(arg)
+
+    index = [in_arr.start_index[i] for i in range(in_arr.rank)]
+
+    for i in range(len(in_arr)):
+        perm = list(permutations(index))
+        for temp_index in perm:
+            res_arr[tuple(index)] += in_arr[tuple(temp_index)]
+        if isinstance(res_arr[tuple(index)], int):
+            res_arr[tuple(index)] = float(res_arr[tuple(index)])
+        res_arr[tuple(index)] /= factorial(in_arr.rank)
+
+        index = in_arr.next_index(index)
+
+    return res_arr
+
+
+def asymmetric(in_arr):
+    """
+    Creates the asymmetric form of input tensor.
+    Input: Arraypy or Tensor with equal axes (array shapes).
+    Output: asymmetric array. Output type - Arraypy or Tensor, depends of input
+
+    Examples
+    ========
+
+    >>> from sympy.tensor.arraypy import list2arraypy
+    >>> from sympy.tensor.tensor_methods import asymmetric
+    >>> a = list2arraypy(list(range(9)), (3,3))
+    >>> b = asymmetric(a)
+    >>> print (b)
+    0  -1.00000000000000  -2.00000000000000
```

Comment:
OK, so maybe arraypy should sympify all of its elements. That is, when assigning elements to the array, do `x = sympify(x)` before storing them.